common/src/components/NotificationsList.tsx

```diff
@@ -0,0 +1,167 @@
+import React from "react";
+import { IconCross, NotificationType } from "hds-react";
+import styled from "styled-components";
+import { useLocalStorage } from "react-use";
+import { useQuery } from "@apollo/client";
+import NotificationWrapper from "./NotificationWrapper";
+import { getTranslation } from "../common/util";
+import { breakpoints } from "../common/style";
+import { fontBold } from "../common/typography";
+import { BannerNotificationType, Query } from "../../types/gql-types";
+import { BANNER_NOTIFICATIONS_LIST } from "./BannerNotificationsQuery";
+
+type NotificationListProps = {
+  target: "USER" | "STAFF";
+};
+
+type NotificationItemProps = {
+  notification: BannerNotificationType;
+  closeFn: React.Dispatch<React.SetStateAction<string[] | undefined>>;
+};
+
+const PositionWrapper = styled.div`
+  width: 100%;
+  display: grid;
+`;
+
+const NotificationBackground = styled.div`
+  position: relative;
+  display: flex;
+  > div {
+    width: 100vw;
+  }
+`;
+
+const NotificationContainer = styled(NotificationWrapper)`
+  font-size: var(--fontsize-body-m);
+  width: calc(100vw - 48px);
+`;
+
+const NotificationText = styled.span`
+  font-size: var(--fontsize-body-m);
+  @media (width < ${breakpoints.xl}) {
+    padding-right: var(--spacing-l);
+  }
+`;
+
+const NotificationDate = styled.span`
+  margin-right: 0.25rem;
+  font-size: var(--fontsize-body-m);
+  ${fontBold}
+`;
+
+const CloseButton = styled.button`
+  position: absolute;
+  top: var(--spacing-m);
+  right: var(--spacing-m);
+  transform: translateY(-50%);
+  padding: var(--spacing-xs);
+  padding-right: 0;
+  background-color: transparent;
+  border: none;
+  cursor: pointer;
+  &:hover {
+    color: var(--color-black-50);
+  }
+`;
+
+const NotificationsListItem = ({
+  notification,
+  closeFn,
+}: NotificationItemProps) => {
+  const displayDate = new Date(notification.activeFrom || "");
+  let notificationType;
+  switch (notification.level) {
+    case "EXCEPTION":
+      notificationType = "alert" as NotificationType;
+      break;
+    case "WARNING":
+      notificationType = "error" as NotificationType;
+      break;
+    default:
+      notificationType = "info" as NotificationType;
```

Comment:
Does `let notificationType: NotificationType = 'info'` or `notificationType = 'error' as const` remove the need for `as` cast?

Reply:
Yep, indeed it does. Much cleaner to declare the type than to cast the variables - cheers!